packages/core/src/awsService/cloudWatchLogs/commands/tailLogGroup.ts

```diff
@@ -32,32 +32,29 @@ export async function tailLogGroup(
         region: wizardResponse.regionLogGroupSubmenuResponse.region,
     }
     const session = new LiveTailSession(liveTailSessionConfig)
-    if (registry.has(session.uri)) {
+    if (registry.has(session.uri.toString())) {
```

Comment:
`uriToKey` exists for this purpose. https://github.com/aws/aws-toolkit-vscode/blob/d4d1ae0d065edc4ac795de32996110efb7dbf0dc/packages/core/src/awsService/cloudWatchLogs/cloudWatchLogsUtils.ts#L33

`toString` is not reliable, querystring parameters can appear in arbitrary order.

Reply:
https://github.com/aws/aws-toolkit-vscode/blob/d4d1ae0d065edc4ac795de32996110efb7dbf0dc/packages/core/src/awsService/cloudWatchLogs/registry/logDataRegistry.ts#L51-L53

This makes me wonder why `LiveTailSessionRegistry` is using a completely different pattern than `LogDataRegistry`. Now we have two different "registries" concepts to maintain, using completely different patterns.

Reply:
I think the aim of these two registries are different from one another.
Given a LiveTail session has its own components unique to other CWL experiences (own client, abortController, responseStream, etc.) We have a `LiveTailSession` object, that has a long running lifeCycle. This lifecycle can end from outside the TailLogGroup command, and so this registry is really just there to track what Sessions are currently running, so when a session needs to be closed via a CodeLens, or closing its textEditors, we can find the LiveTailSession from its `uri` and abort the response stream and clean up its resources. 

I think this is a separate responsiblity from the LogDataRegistry that is actually persisting LogData, and making requests to FilterLogEvents.